neoq.go

```diff
@@ -335,7 +344,14 @@ func execHandler(ctx context.Context, handler Handler) (err error) {
 		err = <-errCh
 		return
 	case <-deadlineCtx.Done():
-		err = fmt.Errorf("job exceeded its %s deadline", handler.deadline)
+		ctxErr := deadlineCtx.Err()
+		if errors.Is(ctxErr, context.DeadlineExceeded) {
+			err = fmt.Errorf("job exceeded its %s deadline", handler.deadline)
```

Comment:
You're losing the ability for downstream consumers to match in the original error type. I'd suggest wrapping the original errors here

Reply:
Good call.